bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/lgservices/api/LGThinqGateway.java

```diff
@@ -0,0 +1,151 @@
+    private String country = "";
+    private String username = "";
+    private String password = "";
+    private String alternativeEmpServer = "";
+    private int accountVersion;
+
+    public LGThinqGateway() {
+    }
+
+    public LGThinqGateway(GatewayResult gwResult, String language, String country, String alternativeEmpServer) {
+        this.apiRootV2 = gwResult.getThinq2Uri();
+        this.apiRootV1 = gwResult.getThinq1Uri();
+        this.loginBaseUri = gwResult.getEmpSpxUri();
+        this.authBase = gwResult.getEmpUri();
+        this.empBaseUri = gwResult.getEmpTermsUri();
+        this.language = language;
+        this.country = country;
+        this.alternativeEmpServer = alternativeEmpServer;
+    }
+
+    @JsonIgnore
+    public String getTokenSessionEmpUrl() {
+        return alternativeEmpServer.isBlank() ? LG_API_V2_EMP_SESS_URL : alternativeEmpServer + LG_API_V2_EMP_SESS_PATH;
+    }
+
+    public String getEmpBaseUri() {
+        return empBaseUri;
+    }
+
+    public int getAccountVersion() {
+        return accountVersion;
+    }
+
+    public String getApiRootV2() {
+        return apiRootV2;
+    }
+
+    public String getAuthBase() {
+        return authBase;
+    }
+
+    public String getLanguage() {
+        return language;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public String getLoginBaseUri() {
+        return loginBaseUri;
+    }
+
+    public String getApiRootV1() {
+        return apiRootV1;
+    }
+
+    public void setEmpBaseUri(String empBaseUri) {
+        this.empBaseUri = empBaseUri;
+    }
+
+    public void setLoginBaseUri(String loginBaseUri) {
+        this.loginBaseUri = loginBaseUri;
+    }
+
+    public void setApiRootV1(String apiRootV1) {
+        this.apiRootV1 = apiRootV1;
+    }
+
+    public void setApiRootV2(String apiRootV2) {
+        this.apiRootV2 = apiRootV2;
+    }
+
+    public void setAuthBase(String authBase) {
+        this.authBase = authBase;
+    }
+
+    public void setLanguage(String language) {
+        this.language = language;
+    }
+
+    public void setCountry(String country) {
+        this.country = country;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    @Override
+    public String toString() {
+        return "LGThinqGateway{" + "empBaseUri='" + empBaseUri + '\'' + ", loginBaseUri='" + loginBaseUri + '\''
+                + ", apiRootV1='" + apiRootV1 + '\'' + ", apiRootV2='" + apiRootV2 + '\'' + ", authBase='" + authBase
+                + '\'' + ", language='" + language + '\'' + ", country='" + country + '\'' + ", username='"
+                + (!username.isEmpty() ? "******" : "<blank>") + '\'' + ", password='"
```

Comment:
username is generally not considered a secret. This should not need to be masked.